src/monmove.cpp

```diff
@@ -101,16 +101,19 @@ bool monster::is_immune_field( const field_type_id fid ) const
 bool monster::can_move_to( const tripoint &p ) const
 {
     const bool can_climb = has_flag( MF_CLIMBS ) || has_flag( MF_FLIES );
-    if( g->m.impassable( p ) && !( can_climb && g->m.has_flag( "CLIMBABLE", p ) ) ) {
+    const bool non_diggable = g->m.has_flag( "ROAD", p ) ||
```

Comment:
This means there are terrains that are diggable (according to this logic yielding `non_diggable = false`), but that are also *not* diggable (according to the absence of the "DIGGABLE" flag). E.g. a terrain that has none of the mentioned flags, like "t_window".

So why is this not simply this?
```suggestion
    const bool non_diggable =  !g->m.has_flag( "DIGGABLE", p );
```

Reply:
Simple `!g->m.has_flag( "DIGGABLE", p )` will permit burrowing creatures to traverse under certain terrain that I don't want them to travel through, such as `concrete floor`. I think it would be wrong to see worms suddenly appearing from under the concrete floor. The same for `t_window` and other terrain without the `DIGGING` flag - I don't want them to materialize on windows and the like.

Reply:
It warrants at least an explanation in the code. Currently readers will wonder why this is not using the aptly named flag "DIGGABLE".

And windows *are* diggable according to your definition (have neither ROAD, nor WALL nor SUPPORTS_ROOF flag), so the monster *can* move onto it.

Reply:
Ok, I'll add a comment, and also think something about windows.